src/js/abstract/base-component.js

```diff
@@ -83,7 +121,31 @@ export default class BaseComponent extends HTMLElement {
     }
 
     if (!this._isConnected) {
+      this._isConnected = true;
+
+      const { constructor: { observedAttributes } } = this;
+
       this.initialClassName = this.className;
+
+      if (Array.isArray(observedAttributes)) {
+        if (ENV !== PROD) {
+          lifecycleLogger(this.logLifecycle)(`\n!!! observedAttributes start -> ${this.nodeName}#${this._id}`);
+        }
+
+        observedAttributes.forEach((attr) => {
+          const key = camelize(attr);
+
+          if (this.hasAttribute(attr)) {
+            const value = getAttribute(this, attr);
+
+            this[key] = value;
```

Comment:
this is a bit dangerous to be honest.. What if we override a DOM property like style or so on?

Reply:
i understand we need it for react thought just want to be sure

Reply:
I agree. We need to document it and could check if a `prop` exists already, if so `throw`